### PR TITLE
[lit-labs/compiler] only compile the valid html tag function imported from lit

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
Issue: https://github.com/lit/lit/issues/189
RFC: https://github.com/lit/rfcs/pull/21
Full prototype PR: https://github.com/lit/lit/pull/3984

### Why

This change exists to restrict compilation down to our `html` tag function. The compiler will only run over a file if an `html` import is detected from `lit` or `lit-html`.

### How

In the pass to locate all the templates, also look for import declarations, and optionally mark the file for compilation if a valid import is found.

### Test plan

Added two goldens.

### Considerations

This change is better than having no handling, but it is still possible to cause a miscompilation if you import `html` from `lit`, and then shadow it with another tag function. There are two options I can think of that remedy this:

 1. Allow the user to customize and override `isLitTemplate`. It would need a different public API.
 2. Invoke TypeScript TypeChecker analysis & check that the declaration came from a valid import. Maybe it's worth the complexity to do this?

### CODE REVIEW

File: packages/labs/compiler/src/lib/template-transform.ts [hunk 1]
```diff
@@ -139,6 +147,28 @@ class CompiledTemplatePass {
...
+      node.importClause?.namedBindings != null &&
+      ts.isNamedImports(node.importClause.namedBindings)
+    ) {
+      const namedBindings = node.importClause.namedBindings;
+      return namedBindings.elements.some(
+        (imp) => ts.isIdentifier(imp.name) && imp.name.text === 'html'
+      );
+    }
+    return false;
+  }
```
Comment (rictic) [inline packages/labs/compiler/src/lib/template-transform.ts:170]: Can we resolve the reference for the template tag function instead? I'm concerned this approach would miscompile:

```ts
import {html} from 'lit';
import * as static from 'lit/static.js';

function foo() {
  const html = static.html;
  return html``;
}
```

Comment (rictic) [reply packages/labs/compiler/src/lib/template-transform.ts:170]: Also, if we go with the currently implemented approach, does it properly handle code like this:

```ts
import {svg as html} from 'lit';
```

Comment (AndrewJakubowicz) [reply packages/labs/compiler/src/lib/template-transform.ts:170]: Yes – those cases will miscompile. I can add resolution.

**Comment by AndrewJakubowicz** [comment]
Converted to draft – making the checking much more robust.

File: packages/labs/compiler/src/lib/template-transform.ts (hunk 1 above)
Comment (AndrewJakubowicz) [reply packages/labs/compiler/src/lib/template-transform.ts:170]: Also handle:
```
import * as lit from 'lit';

lit.html`rar`;
```

**Comment by AndrewJakubowicz** [comment]
From the great feedback, this PR has been replaced with https://github.com/lit/lit/pull/4055

**Pull request closed without merging**